chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/FilterController.java

```
@@ -0,0 +1,108 @@
+        FilterResults results = new FilterResults();
+        
+        FilterTypeDTO filterTypeDTO = filterManager.getFilterType(filterTypeId);
+        
+        List<Filter> filters = new ArrayList<Filter>();
+        List<FilterDTO> dtos = filterManager.getByFilterType(filterTypeDTO);
+        for (FilterDTO dto : dtos) {
+            filters.add(new Filter(dto));
+        }
+        
+        results.setResults(filters);
+        
+        return results;
+    }
+    
+    @ApiOperation(value = "Save filter for the current user.",
+            notes = "")
+    @RequestMapping(value = "", method = RequestMethod.POST, produces = "application/json; charset=utf-8")
+    public @ResponseBody Filter create(@RequestBody final Filter filter) throws EntityRetrievalException, UserRetrievalException, ValidationException {
+        FilterDTO dto = new FilterDTO();
+        UserDTO userDTO = userManager.getById(Util.getCurrentUser().getId());
+        FilterTypeDTO filterTypeDTO = filterManager.getFilterType(filter.getFilterType().getId());
+        
+        dto.setName(filter.getName());
+        dto.setFilter(filter.getFilter());
+        dto.setUser(userDTO);
+        dto.setFilterType(filterTypeDTO);
+        
+        dto = filterManager.create(dto);
+        
+        return new Filter(dto);
+    }
+    
+    @ApiOperation(value = "Deletes a filter.",
+            notes = "")
+    @RequestMapping(value = "/{filterId}", method = RequestMethod.DELETE,
+    produces = "application/json; charset=utf-8")
+    public @ResponseBody String deleteFilter(@PathVariable("filterId") final Long filterId) {
+        try {
+            FilterDTO filterDTO = filterManager.getByFilterId(filterId);
+            if (filterDTO != null) {
+                filterManager.delete(filterDTO);
+                return "{\"success\" : true}";
+            } else {
+                return "{\"success\" : false}";
```

I think if the code executes this line it would mean that the filterId passed in wasn't valid and I think we should return an error code in that case. Bad Request? Maybe either throw an exception here that will be caught and handled in that way by the controller advice class or just send back the Bad Request status code yourself here.